app/src/main/java/com/adg/events/droidcon_app/ioc/DroidconAppModule.kt

```diff
@@ -0,0 +1,28 @@
+package com.adg.events.droidcon_app.ioc
+
+import android.content.Context
+import com.adg.events.droidcon_app.DroidconApp
+import com.droidcon.commons.data.db.AppDatabase
+import com.droidcon.commons.data.db.SessionsDao
+import dagger.Module
+import dagger.Provides
+import javax.inject.Singleton
+
+
+@Module
+class DroidconAppModule {
+
+    @Provides
+    fun provideContext(application: DroidconApp): Context {
+        return application.applicationContext
+    }
+
+    @Singleton
+    @Provides
+    fun providesAppDatabase(context: Context): AppDatabase = AppDatabase.getDatabase(context)
```

Comment:
As talked, we'll move this to :commons module and also the repos because they're going to be used by different modules (e.g. schedule -> speakers or speakers -> schedule)

Reply:
I agree with moving repos to commons module but I am not sure we can move this database provider as room database initialization requires context, which is something not available in commons module

Reply:
There is no problem with the access to app context from :commons, you can check how it would look like in the next link https://github.com/ADGevents/android-droidcon-madrid-19/commit/3a48210de60ecb6fa3ed8502bdc52a3fa46397e8#diff-f535ef0b7f898f212602831355d5a8a8